src/app/contact-view/contact-view.component.ts

```diff
@@ -0,0 +1,31 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription, switchMap } from 'rxjs';
+import {ContactService} from "../Contact/contact.service";
+import {Contact} from "../Contact/contact";
+@Component({
+  selector: 'app-contact-view',
+  templateUrl: './contact-view.component.html',
+  styleUrls: ['./contact-view.component.css']
+})
+export class ContactViewComponent implements OnInit, OnDestroy {
+  contact: Contact | undefined;
+
+  constructor(private readonly route: ActivatedRoute, private contactService: ContactService) {}
+
+  id$ = this.route.params.pipe(switchMap((params) => params['id']));
+
+  async ngOnInit() {
+    this.route.params.subscribe(params => {
```

Comment:
This will still create a subscription. I can see that you're using the observable method of getting the id - I would remove this subscribe call.